src/app/qgisapp.cpp

```diff
@@ -16060,6 +16060,18 @@ bool QgisApp::addRasterLayers( QStringList const &files, bool guiWarning )
         continue;
     }
 
+    const bool isVsiCurl { src.startsWith( QLatin1String( "/vsicurl" ), Qt::CaseInsensitive ) };
+    const auto scheme { QUrl( src ).scheme() };
+    const bool isRemoteUrl { src.startsWith( QLatin1String( "http" ) ) || src == QLatin1String( "ftp" ) };
+
+    std::unique_ptr< QgsTemporaryCursorOverride > cursorOverride;
```

Comment:
You'll need to manually clear this somewhere, like the vector code path does (ie. before we could show any dialogs such as in addLayerPrivate)